### ISSUE TITLE
Invalid warning for .attr('selected') with element creation (1.4.1)

### ISSUE THREAD

**Issue by adrian-sieracki** [jquery/jquery-migrate#465]
Sample element creation with jQuery where default properties are provided:
`$('<option>', {
  value: "...",
  selected: true
})`

This of course contains 'value' and 'selected' which should not be used with attr(). However in this situation they are used as attributes to define default values of an element where usage of attr() is correct. More over that's not used by user code but it's internal jQuery (2.2.4) call.

Trace:
`migrateWarn	@	jquery-migrate.js:43
jQuery.attr	@	jquery-migrate.js:159
access	@	jquery.js:3628
attr	@	jquery.js:7116
jQuery.fn.init	@	jquery.js:2882
jQuery.fn.init	@	jquery-migrate.js:241
jQuery	@	jquery.js:75`

jQuery calls init() for element creation and uses there attr() for setting up HTML attributes with default values:
`// HANDLE: $(html, props)
if ( rsingleTag.test( match[ 1 ] ) && jQuery.isPlainObject( context ) ) {
  for ( match in context ) {
  
	  // Properties of context are called as methods if possible
	  if ( jQuery.isFunction( this[ match ] ) ) {
		  this[ match ]( context[ match ] );
  
	  // ...and otherwise set as attributes
	  } else {
		  this.attr( match, context[ match ] );
	  }
  }
}`

this.attr( match, context[ match ] ); -> leads to attr() from jQuery migrate which doesn't check such situations:
`// Warn only for attributes that can remain distinct from their properties post-1.9
if ( ruseDefault.test( lowerName ) ) {
	migrateWarn( "jQuery.fn.attr('" + lowerName + "') might use property instead of attribute" );
}`

In summary this causes incorrect warning notification which pollutes my final results when investigating for application for migration compatibilities issues. That's especially annoying when custom notification solution is implemented to replace console output and provide this for non-technical users to support validation process.

**Comment by adrian-sieracki** [comment]
Additional impact of that incorrect warning:
jQuery.migrateWarnings stores unique warnings and only first occurrences are reported to the console. As this is not fixable by application code and in my case it appears as first call then any other occurrences are hidden and this complicates further investigation.

**Comment by dmethvin** [comment]
Any time an attribute is reflected into a property, the property should be set on the DOM element. So the warning seems correct?

The `$(elem, { attr: value})` form is dangerous since at runtime it looks at whether there is a method with that name. If someone adds a `selected` plugin it could break your code. Or who knows, it might eliminate the warning if the plugin sets the selected _property_ on the element. Also, there are some edge cases where setting `value` attr may not work properly, which is why there is the `.val()` method.

For all those reasons, changing the code to `$('<option>').val(...).prop({selected: true })` seems best.

**Comment by dmethvin** [comment]
Also, I vaguely recall we deprecated the `$(elem, { attr: value})` form because it is error-prone. Since `jQuery()` has so many signatures it would be hard to tell in the docs.
